python/grass/script/core.py

```diff
@@ -1709,44 +1718,38 @@ def create_location(
     desc=None,
     overwrite=False,
 ):
-    """Create new location
+    """Create new project
 
     Raise ScriptError on error.
 
-    :param str dbase: path to GRASS database
-    :param str location: location name to create
-    :param epsg: if given create new location based on EPSG code
-    :param proj4: if given create new location based on Proj4 definition
-    :param str filename: if given create new location based on georeferenced file
-    :param str wkt: if given create new location based on WKT definition
+    :param str path: path to GRASS database
+    :param str name: project name to create
+    :param str mapset: mapset within given project (default: 'PERMANENT')
+    :param epsg: if given create new project based on EPSG code
+    :param proj4: if given create new project based on Proj4 definition
+    :param str filename: if given create new project based on georeferenced file
+    :param str wkt: if given create new project based on WKT definition
                     (can be path to PRJ file or WKT string)
     :param datum: GRASS format datum code
     :param datum_trans: datum transformation parameters (used for epsg and proj4)
-    :param desc: description of the location (creates MYNAME file)
-    :param bool overwrite: True to overwrite location if exists(WARNING:
-                           ALL DATA from existing location ARE DELETED!)
+    :param desc: description of the project (creates MYNAME file)
+    :param bool overwrite: True to overwrite project if exists(WARNING:
+                           ALL DATA from existing project ARE DELETED!)
     """
-    # create dbase if not exists
-    if not os.path.exists(dbase):
-        os.mkdir(dbase)
+    # Support ~ in the path for user home directory.
+    path = Path(path).expanduser()
 
-    # check if location already exists
-    if os.path.exists(os.path.join(dbase, location)):
-        if not overwrite:
-            warning(_("Location <%s> already exists. Operation canceled.") % location)
-            return
-        else:
-            warning(
-                _("Location <%s> already exists and will be overwritten") % location
-            )
-            shutil.rmtree(os.path.join(dbase, location))
+    if name:
+        mapset = "PERMANENT"
+    else:
+        path = path / "PERMANENT"
+        mapset = None
```

Comment:
It looks like `resolve_mapset_path` already does this, so we should not duplicate this.

Reply:
I think this does need to be here. `resolve_mapset_path` expects the mapset somewhere in the keyword arguments so we have to add it here before passing to `resolve_mapset_path`. Otherwise, if you pass `path="~/grassdata/project_name"`, `resolve_mapset_path` will return:
```
mapset_path.directory = "~"
mapset_path.location = "grassdata"
mapset_path.mapset = "project_name"
```

Reply:
Good point, how about this (I haven't tested):
```
if not name:
    path = path / "PERMANENT"
mapset_path = resolve_mapset_path(path=path, location=name)
```

also

Reply:
Yes, that works. I tested. Added it in latest commit.